### DISCUSSION TITLE
Custom Filters

### DISCUSSION THREAD

**Discussion by BenOussama180** [Power-Components/livewire-powergrid#748] [Ideas]
maybe custom filters can become a thing ? because right now we can only filter by existing columns that we have inside the database, i want to make to make a custom filter that foor exwample either returns true or false for each record and returns the "true" records
foor example in my case scenario : 
i have a table of Links  , and each link can be related to 0 or multiple projects depending on the auth user, so i want to show a boolean filter in the links table for " Related to a project " with yes or no option.
is this achievable with powergrid's current state ?

**Comment by thedangler** [comment]
I need this too. If I have a relationship in the table I need to show True if not False.
And I need them to filter on Yes or no.

**Reply by BenOussama180** [reply]
i managed to make it work with a custom attribute and passed that custom attribute to the boolean filter and it worked

**Reply by thedangler** [reply]
Can you show me the code?
I tried that but It did process my condition to show Yes or No properly like my addColumn.

**Reply by BenOussama180** [reply]
check my comment under

**Comment by BenOussama180** [comment]
![image](https://user-images.githubusercontent.com/81111919/206175554-c7027b39-a6f9-4521-9034-af0bad4f05f8.png)
![image](https://user-images.githubusercontent.com/81111919/206175590-f98653d5-e960-4179-ba22-05a76398765b.png)
and in your Model : 
![image](https://user-images.githubusercontent.com/81111919/206175639-ca138bfb-996d-4063-a458-e30e11121d33.png)
![image](https://user-images.githubusercontent.com/81111919/206175655-03de2aa3-af5f-4281-830c-d4313f41a5b1.png)
basically i just check if link has been bought by the auth user and i show a green or red mark with some logic on hover u dont have to worry about it , and i filter by that custom proprety that i made and i append it to the model for better performance its like eager loading

**Reply by BenOussama180** [reply]
Bare in mind that you datasource need to return a collection for this to work, because if you dont call the `->get()` your model wont be having the custom attribute, and filtering collections is pretty slow so i wont advice you to do this approach for large databases
